lib/fog/vcloud_director/generators/compute/compose_common.rb

```diff
@@ -187,6 +197,55 @@ def array_wrap(val)
             return val if val.kind_of?(Array)
             [val].compact
           end
+
+          def vapp_network_section(xml, name:, subnet:, description: nil, deployed: nil, parent: nil, parent_name: nil,
+                                   fence_mode: nil, retain: false, external_ip: nil)
+            description ||= name
+            parent = href("/network/#{parent}") if parent
+            fence_mode ||= 'bridged' if parent
```

Comment:
Is it ok that the user is allowed to set `fence_mode`? Here, it looks to me that if `parent` is set, it should be `bridge`. But if the user has submitted `isolated` and the `parent` is also set, it will remain `isolated`.

Will such network be unusable? If so, would it perhaps make sense to override the user's `fence_mode`?

I am asking because in L208 you then have a more concrete test, so I wonder if 205-208 could be simplified?